src/connection.c

```diff
@@ -30,24 +30,19 @@
 static ConnectionType *connTypes[CONN_TYPE_MAX];
 
 int connTypeRegister(ConnectionType *ct) {
-    const char *typename = ct->get_type(NULL);
-    ConnectionType *tmpct;
-    int type;
-
-    /* find an empty slot to store the new connection type */
-    for (type = 0; type < CONN_TYPE_MAX; type++) {
-        tmpct = connTypes[type];
-        if (!tmpct) break;
+    int type_id = ct->get_type_id(NULL);
+    if (type_id <= 0 || type_id >= CONN_TYPE_MAX) {
+        serverLog(LL_WARNING, "Invalid connection type id %d", type_id);
+        return C_ERR;
+    }
```

Comment:
a serverAssert seems better to me.

```suggestion
    serverAssert(type_id > 0 && type_id < CONN_TYPE_MAX);
```